packages/@aws-cdk/aws-eks/test/test.awsauth.ts

```diff
@@ -71,31 +71,62 @@ export = {
                 "Arn"
               ]
             },
-            "\\\",\\\"groups\\\":[\\\"role-group2\\\",\\\"role-group3\\\"]}]\",\"mapUsers\":\"[{\\\"userarn\\\":\\\"arn:",
+            "\\\",\\\"groups\\\":[\\\"role-group2\\\",\\\"role-group3\\\"]}]\",\"mapUsers\":\"[{\\\"userarn\\\":\\\"",
             {
-              Ref: "AWS::Partition"
+              "Fn::GetAtt": [
+                "user2C2B57AE",
+                "Arn"
+              ]
             },
-            ":iam:us-east-1:",
+            "\\\",\\\"groups\\\":[\\\"user-group1\\\",\\\"user-group2\\\"]},{\\\"userarn\\\":\\\"",
             {
-              Ref: "AWS::AccountId"
+              "Fn::GetAtt": [
+                "user2C2B57AE",
+                "Arn"
+              ]
             },
-            ":user/",
+            "\\\",\\\"username\\\":\\\"foo\\\",\\\"groups\\\":[\\\"user-group1\\\",\\\"user-group2\\\"]}]\",\"mapAccounts\":\"[\\\"112233\\\",\\\"5566776655\\\"]\"}}]"
+          ]
+        ]
+      }
+    }));
+
+    test.done();
+  },
+
+  'imported users and roles can be also be used'(test: Test) {
+    // GIVEN
+    const { stack } = testFixtureNoVpc();
+    const cluster = new Cluster(stack, 'Cluster');
+    const role = iam.Role.fromRoleArn(stack, 'imported-role', 'arn:aws:iam::123456789012:role/S3Access');
+    const user = iam.User.fromUserName(stack, 'import-user', 'MyUserName');
+
+    // WHEN
+    cluster.awsAuth.addRoleMapping(role, { groups: ['group1'] });
+    cluster.awsAuth.addUserMapping(user, { groups: ['group2'] });
+
+    // THEN
+    expect(stack).to(haveResource(KubernetesResource.RESOURCE_TYPE, {
+      Manifest: {
+        "Fn::Join": [
+          "",
+          [
+            "[{\"apiVersion\":\"v1\",\"kind\":\"ConfigMap\",\"metadata\":{\"name\":\"aws-auth\",\"namespace\":\"kube-system\"},\"data\":{\"mapRoles\":\"[{\\\"rolearn\\\":\\\"",
```

Comment:
Surely this could be encoded in single quotes so that this escaping madness is not necessary?

Reply:
I intentionally pasted the test output so that it will be easy to update if needed in the future without needing to manually replace quotes